src/org/sosy_lab/common/collect/OrderStatisticSet.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *  SoSy-Lab Common is a library of useful utilities.
+ *  This file is part of SoSy-Lab Common.
+ *
+ *  Copyright (C) 2007-2015  Dirk Beyer
+ *  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.sosy_lab.common.collect;
+
+import java.util.SortedSet;
+
+public interface OrderStatisticSet<T> extends SortedSet<T> {
```

Comment:
The interface and the methods should have JavaDoc that explains the guarantees.

Why is the name `OrderStatisticSet`? Why not `SortedUniqueList` or something else?

Reply:
Should I really add performance guarantees to the interface?
Despite that, JavaDoc is obviously missing, of course. I added it.

* Why not `SortedUniqueList`?

Lists provide more methods for indexed access (e.g., to add a value at a specific position),
so I thought that any interface name with `List ` in it may be misleading.
In addition, it would be easily possible to add a *MultiSet and *Map interface and implementation with similar capabilities, so the `Unique` part would have to be removed
and the new interfaces would have to get different names, reducing the 'grouping' of such interfaces.

* Why `OrderStatistic`?

It's not used in any interface description yet, we could easily add `OrderStatisticMultiset` and `OrderStatisticMap` interfaces, and the [meaning](https://en.wikipedia.org/wiki/Order_statistic) is correct: The interface allows 1) to get the k-th smallest value of a set and 2) to determine the k for a given value.
In addition, a data structure [order statistic tree](https://en.wikipedia.org/wiki/Order_statistic_tree) already exists in literature that provides the same capabilities.

That said, I'm open for names that are more self-explanatory, I just couldn't find one yet.

Reply:
Ok, if the name is already well-known, let's use it.

For guarantees in JavaDoc, I meant all kinds of guarantees, but performance would also be interesting if possible and meaningful for the interface (we could look at `Set`, `List`, etc., if they provide performance guarantees). Otherwise only `SkipList` should document performance guarantees.

Reply:
The collection interfaces don't provide performance guarantees. But could  you have a look at the new JavaDoc to check whether you miss any information, please?